src/main/java/org/xerial/snappy/SnappyInputStream.java

```diff
@@ -331,16 +356,24 @@ protected boolean hasNextChunk() throws IOException {
         uncompressedCursor = 0;
         uncompressedLimit = 0;
 
-        int readBytes = 0;
-        while (readBytes < 4) {
-            int ret = in.read(chunkSizeBuf, readBytes, 4 - readBytes);
-            if (ret == -1) {
-                finishedReading = true;
+        int readBytes = readNext(header, 0, 4);
+        if(readBytes < 4)
+            return false;
+
+        int chunkSize = SnappyOutputStream.readInt(header, 0);
+        if(chunkSize == SnappyCodec.MAGIC_HEADER_HEAD) {
+            // Concatenated data
+            int remainingHeaderSize = SnappyCodec.headerSize() - 4;
+            readBytes = readNext(header, 4, remainingHeaderSize);
+            if(readBytes < remainingHeaderSize)
```

Comment:
good point. We should add coverage report (jacoco, etc.) and some coding style enforcer to use curly braces for single statement 'if' blocks.